### ISSUE TITLE
Uninformative Page Title 

### ISSUE THREAD

**Issue by johnhbenetech** [benetech/MathShare#743]
While navigating through and using the MathShare application, the page title always remains the same (“Benetech Mathshare”).  This means that users are unable to: 

* determine which content is being displayed from the browser title bar alone;
* quickly distinguish between multiple browser tabs which are all showing MathShare content without reading at least some of that content; or
* search for MathShare content in their browser history or bookmarks.

#### Suggested Solution 
 
Update the global `document.title` variable, either when new routes are loaded or significant content changes occur.  Page titles should be informative but concise.  For example:

* For the `/app` route:
    * Good title: “All Problem Sets - Benetech Mathshare”
    * Bad title: “Problems”
* For `/app/problemSet/solve/TX45BTPJBIPJY`:
    * Good title: “Distance, Rate and Time Problems - Benetech Mathshare”
    * Bad title: “Problem Set”
* For `/app/problem/edit/74ZDMXS5Z3W22`:
    * Good title: “Problem 1 of 3 | Distance, Rate and Time - Benetech Mathshare”
    * Bad title (too long): “An airplane moves at a constant speed of 120 miles per hour for 3 hours. How far does it go?: 120 miles slash hour comma 3 hours comma solve for x open paren distance close paren”

These are only samples.  Appropriate titles for other routes (both current and future) should be determined.  Some suggestions for creating effective page titles are below.

* Follow a most-to-least significance order, placing information which is likely to change more frequently closer to the beginning.  For example, “Problem 1 of 3 | Distance, Rate and Time - Benetech Mathshare” is a better title than “Benetech Mathshare - Distance, Rate and Time | Problem 1 of 3”.
* A good pattern to follow is: “`[page name (if present)] | [website subsection (if present)] - [site name]`”.

Aha! Link: https://diagramlabs.aha.io/features/B-135

**Comment by rupeshparab** [comment]
@johnhbenetech 
I have set these 3 mentioned titles, can you please ask someone from the product team to decide the titles for the other pages too?

**Comment by johnhbenetech** [comment]
@benetechMatt @alexrcabral can you give us some titles for the various pages?

**Comment by alexrcabral** [comment]
Sure thing. I think I have all separate pages covered including Sina's suggestions, please let me know if I've missed anything.

Overview - Benetech Mathshare
https://mathshare.benetech.org/#/

Partners - Benetech Mathshare 
https://mathshare.benetech.org/#/partners

Policies - Benetech Mathshare 
https://mathshare.benetech.org/#/privacy

Open Source Partnerships - Benetech Mathshare 
https://mathshare.benetech.org/#/partners

Sign In - Benetech Mathshare 
https://mathshare.benetech.org/#/signIn

All Problem Sets - Benetech Mathshare
https://mathshare.benetech.org/#/app

Tutorial - Benetech Mathshare 
https://mathshare.benetech.org/#/app/problem/example

[Problem Set Title] - Benetech Mathshare 
[any main problem set page] 

Problem [current problem#] of [total problem #] | [Problem Set Title] - Benetech Mathshare 
[any page on an individual problem]

**Comment by johnhbenetech** [comment]
Back to you @rupeshparab 

Thanks @alexrcabral

**Comment by rupeshparab** [comment]
> Open Source Partnerships - Benetech Mathshare
> https://mathshare.benetech.org/#/partners

2 titles were specified for  - 'Open Source Partnerships - Benetech Mathshare' and 'Partners - Benetech Mathshare'
I am using 'Partners - Benetech Mathshare'

**Comment by jscholes** [comment]
@rupeshparab - there is no page title for [/app/problemSet/solve/WU2ICGFB525TI](https://mathshare-qa.diagramcenter.org/#/app/problemSet/solve/WU2ICGFB525TI) (click on "Example Problem" from the "All Problem Sets" page).

**Comment by benetechMatt** [comment]
I would suggest "Untitled Problem Set"

**Comment by johnhbenetech** [comment]
@benetechMatt I assume we want 'Example Probelm Set' for the statically defined example we provide. then yes 'Untitled Problem Set' if someone summons a problem set from before we had titles.

**Comment by benetechMatt** [comment]
@johnhbenetech, correct, the one set that is title "example problem set" will keep that title.  Other problems without a title will get the "untitled problem set" and all new problem sets will receive the current style of "New Problem Set [today's date]"

**Comment by rupeshparab** [comment]
I have updated the page titles

**Comment by jscholes** [comment]
@rupeshparab - when a problem set has no title, the page title is correct but the main level 1 heading text is blank.  The heading should also have the "Untitled Problem Set" title.

**Comment by jscholes** [comment]
Fixed in https://github.com/benetech/MathShare/commit/b4044d268bc7de88276f2324504cf4a164158cd1.
